File: backend/app/schemas/live.py
from pydantic import BaseModel, Field


class LiveAuthTokenRequest(BaseModel):
    model: str | None = Field(default=None, description="Optional live model override.")
    modalities: list[str] | None = Field(default=None, description="Requested response modalities.")
    voice: str | None = Field(default=None, description="Optional voice override.")
    template_code: str | None = Field(default=None, description="Prompt template code.")
    system_instruction: str | None = Field(default=None, description="Optional raw system instruction.")


class LiveAuthTokenResponse(BaseModel):
    auth_token: str = Field(..., description="Ephemeral auth token resource name.")
    model: str = Field(..., description="Resolved Gemini Live model.")
    modalities: list[str] = Field(..., description="Effective response modalities.")
    voice: str | None = Field(default=None, description="Effective voice name.")
    template_code: str | None = Field(default=None, description="Resolved prompt template code.")
    system_instruction: str | None = Field(default=None, description="Resolved system instruction.")
    display_endpoint: str = Field(..., description="User-facing description of the client-to-server route.")

File: frontend/src/api/live.ts
import { z } from 'zod';

import { http } from './http';

const LiveAuthTokenResponseSchema = z.object({
  auth_token: z.string(),
  model: z.string(),
  modalities: z.array(z.string()),
  voice: z.string().nullable().optional(),
  template_code: z.string().nullable().optional(),
  system_instruction: z.string().nullable().optional(),
  display_endpoint: z.string(),
});

export type LiveAuthTokenPayload = {
  model?: string;
  modalities?: string[];
  voice?: string;
  templateCode?: string;
  systemInstruction?: string;
};

export type LiveAuthTokenResponse = z.infer<typeof LiveAuthTokenResponseSchema>;

export async function fetchLiveAuthToken(
  payload: LiveAuthTokenPayload
): Promise<LiveAuthTokenResponse> {
  const response = await http.post('/live/auth-token', {
    model: payload.model,
    modalities: payload.modalities,
    voice: payload.voice,
    template_code: payload.templateCode,
    system_instruction: payload.systemInstruction,
  });

  return LiveAuthTokenResponseSchema.parse(response.data.data);
}
